### DISCUSSION TITLE
Parameter definitions

### DISCUSSION THREAD

**Discussion by mets18** [GerryDazoo/Slinger#272] [General]
Hi, loving this program and it works great.  I am not an experienced streamer and I was fooling around with the parameters in the config.ini file.  I understand what resolution, framerate and bandwidth are but was curious about IframeRate and VideoSmoothness.  Is higher better?  Thanks.

**Comment by RandomMan719** [comment]
IframeRate is how often it sends and I frame, which is a part of Mpeg encoding, I frames contain all the information for one frame.  B & P frames only contain information that changes from frame to frame.  The higher IframeRate, will use more data to stream because it is sending more information.  I am not sure what VideoSmoothness is, I have messed with settings on both as well.  I ended up just leaving them at default.  It would be nice if the settings could be adaptive like the Slingplayer, where they change automatically with available bandwidth.

**Reply by GerryDazoo** [reply]
Unfortunately we cannot be adaptive like slingplayer because there is no way to get any feedback from the media player you are using  about the video quality. It works for Sling because they are the server and the player.

**Reply by DoogieHowitzerWMD** [reply]
> Unfortunately we cannot be adaptive like slingplayer because there is no way to get any feedback from the media player you are using about the video quality. It works for Sling because they are the server and the player.

I hadn't thought about that interworking aspect of it, but that makes total sense based on what I'm seeing. That bit of occasional stuttering I see with the streams from my Slinger server is something that won't adaptively self-stabilize.

I'm wondering whether it would help to specify even larger TCP buffer sizes for the server than what is specified in the readme, as well as larger cache sizes in the media player. Maybe to a certain extent?

The bigger question is if there is a way that the latency itself could be increased for the sake of gains in stream stability without sacrificing video quality? Personally, I wouldn't really care if it took a solid 10 to 15 seconds before I see the result of remote code actions, if the trade-off is that I'm running 720p @ 60fps while dealing with 10% of the amount of stuttering.
